test/test_fx.py

```diff
@@ -1769,7 +1769,106 @@ def forward(self, x: torch.Tensor):
         graph = ast_rewriter.trace(M())
         traced = GraphModule(ast_rewriter.root, graph, "gm")
 
-        traced.graph.lint(traced)
+        traced.graph.lint()
+
+    def test_submodule_manipulation_API(self):
+        class C(torch.nn.Module):
+            def __init__(self):
+                super(C, self).__init__()
+                self.conv = torch.nn.Conv2d(16, 33, 3, stride=2)
+
+            def forward(self, x):
+                return self.conv(x)
+
+        class B(torch.nn.Module):
+            def __init__(self):
+                super(B, self).__init__()
+                self.linear = torch.nn.Linear(100, 200)
+                self.net_c = C()
+
+            def forward(self, x):
+                return self.linear(self.net_c(x))
+
+        class A(torch.nn.Module):
+            def __init__(self):
+                super(A, self).__init__()
+                self.net_b = B()
+
+            def forward(self, x):
+                return self.net_b(x)
+
+        a = symbolic_trace(A())
+
+        a.insert_submodule("net_b.net_c.dropout", torch.nn.Dropout(p=0.2))
+
+        conv = [n for n in a.graph.nodes if n.target == "net_b.net_c.conv"][-1]
+        with a.graph.inserting_before(conv):
+            dropout = a.graph.call_module(module_name='net_b.net_c.dropout',
+                                          args=conv.args)
+
+        conv.replace_all_uses_with(dropout)
+        a.graph.erase_node(conv)
+        a.recompile()
+
+        def module_exists(gm: GraphModule, path: str) -> bool:
+            atoms: List[str] = path.split(".")
+
+            for item in atoms:
+
+                if not hasattr(gm, item):
+                    return False
+
+                gm = getattr(gm, item)
+
+                if not isinstance(gm, torch.nn.Module):
+                    return False
+
+            return True
+
+
+        # Test that we added the "dropout" submodule
+        self.assertEqual(module_exists(a, "net_b.net_c.dropout"), True)
```

Comment:
übernit: the `unittest` base class has special `assert` methods for checking if things are true, false, etc: https://docs.python.org/3/library/unittest.html#assert-methods